package/MDAnalysis/transformations/boxdimensions.py

```diff
@@ -0,0 +1,82 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""\
+Set box dimensions --- :mod: `MDAnalysis.transformations.boxdimensions`
+=======================================================================
+
+Set dimensions of the simulation box to a constant vector across all timesteps.
+
+
+..autoclass:: setdimensions
+"""
+import numpy as np
+
+
+class setdimensions:
+    """
+    Set simulation box dimensions.
+
+    Timestep dimensions are modified in place.
+
+    Example
+    -------
+
+    e.g. set simulation box dimensions to a vector containing unit cell
+    dimensions [*a*, *b*, *c*, *alpha*, *beta*, *gamma*], lengths *a*,
+    *b*, *c* are in the MDAnalysis length unit (Å), and angles are in degrees.
+
+    .. code-block:: python
+
+        dim = [2, 2, 2, 90, 90, 90]
+        transform = mda.transformations.boxdimensions.setdimensions(dim)
+        u.trajectory.add_transformations(transform)
+
+    Parameters
+    ----------
+    dimensions: array-like
+        vector that contains unit cell lengths and angles.
+        Expected shapes are (6, 0) or (1, 6)
+
+    Returns
+    -------
+    :class:`~MDAnalysis.coordinates.base.Timestep` object
+    """
+
+    def __init__(self, dimensions):
+        self.dimensions = dimensions
+
+        try:
+            self.dimensions = np.asarray(self.dimensions, np.float32)
+        except ValueError:
+            raise ValueError(f'{self.dimensions} cannot be converted into\
+np.float32 numpy.ndarray')
```

Comment:
I believe these line breaks are also going to throw a FutureWarning (also some flake8 things which are less relevant).

Something like this would be better here (setting errmsg is just my own stylistic preference, you can just put the string contents using that line break straight in the ValueError instead).

```suggestion
            errmsg = (f"{self.dimensions} cannot be converted into "
                             "np.float32 numpy.ndarray")
            raise ValueError(errmsg)
```